bin/comment_count.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+if [ "$#" -eq 0 ]; then
+  echo "Pass at least one directory or file" >&2
+  exit 1
+fi
+mapfile -t csv < <( cloc --csv --quiet "$@" 2>/dev/null)
```

Comment:
From the docs, this program is described as counting the number of comments, however,  `cloc` counts number of _comment lines_ in a file and this is an important distinction to make in the docs at least. For example, given the cpp code block:

```cpp
/*
this is a multiline
comment
that goes for a few lines
*/

// this is a single line comment
```

`cloc` will return a comment line count of `6`. If we explicitly want the number of comments vs comment line count, then I think we'll need another tool.

Reply:
I went over some [similar counting tools](https://github.com/AlDanial/cloc#other-counters-). They dont seem to have a functionality to match our use case. 
What I could see as an option is to get the cloc tool's output and seperately go over the source file and counting multi-line comments, and substract the values. (multi-line comments in most of the languages seems to use `/* */`. So effectively check for these symbols) Does that sound as a feasible option.

Reply:
I proposed this idea in the meeting but it didn't seem to go anywhere: What about using tree-sitter to count comments?  We'd be limited by the number of languages tree-sitter supports, but I don't expect that to be an issue given the courses which would likely use this utility the most.  It would take a few dozen lines of code at most to support basic comment-counting functionality, and would allow us to change the behavior as necessary.

Reply:
Tested this using `tree-sitter` it seems to be possible to do this.